mmtrack/evaluation/metrics/tao_metrics.py

```diff
@@ -329,19 +412,83 @@ def _format_one_video_gts(self, gt_dicts: Tuple[dict]) -> List:
 
         return annotations
 
-    def save_pred_results(self, pred_results: List) -> None:
+    def save_pred_results(self, pred_results: List, res_type: str) -> None:
         """Save the results to a zip file.
 
         Args:
             pred_results (list): Testing results of the
                 dataset.
+            res_type (str): The type of testing results, track or detection.
         """
         logger: MMLogger = MMLogger.get_current_instance()
         if self.outfile_prefix is None:
             tmp_dir = tempfile.TemporaryDirectory()
             outfile_prefix = osp.join(tmp_dir.name, 'results')
         else:
             outfile_prefix = self.outfile_prefix
-        mmcv.dump(pred_results, f'{outfile_prefix}.json')
+        mmcv.dump(pred_results, f'{outfile_prefix}_{res_type}.json')
+
+        logger.info(f'save the results to {outfile_prefix}_{res_type}.json')
+
+    def evaluate(self, size: int) -> dict:
+        """Evaluate the model performance of the whole dataset after processing
+        all batches.
+
+        Args:
+            size (int): Length of the entire validation dataset.
+
+        Returns:
+            dict: Evaluation metrics dict on the val dataset. The keys are the
+            names of the metrics, and the values are corresponding results.
+        """
+        # wait for all processes to complete prediction.
+        barrier()
+
+        if len(self.results) == 0:
+            warnings.warn(
+                f'{self.__class__.__name__} got empty `self.results`. Please '
+                'ensure that the processed results are properly added into '
+                '`self.results` in `process` method.')
+
+        results = collect_tracking_results(self.results, self.collect_device)
+
+        # gather seq_info
+        gathered_seq_vids_info = all_gather_object(
+            self._tao_meta_info['videos'])
+        gathered_seq_imgs_info = all_gather_object(
+            self._tao_meta_info['images'])
+        gathered_seq_tracks_info = all_gather_object(
+            self._tao_meta_info['tracks'])
+        all_seq_vids_info = []
+        all_seq_imgs_info = []
+        all_seq_tracks_info = []
+        for _seq_info in gathered_seq_vids_info:
+            all_seq_vids_info.extend(_seq_info)
+        for _seq_info in gathered_seq_imgs_info:
+            all_seq_imgs_info.extend(_seq_info)
+        for _seq_info in gathered_seq_tracks_info:
+            all_seq_tracks_info.extend(_seq_info)
+
+        # update self._tao_meta_info
+        self._tao_meta_info = dict(
+            videos=all_seq_vids_info,
+            images=all_seq_imgs_info,
+            tracks=all_seq_tracks_info)
```

Comment:
Can we save `self._tao_meta_info` into the return value of `_format_one_video_gts()`.

By doing this, we do not need to gather `self._tao_meta_info`, since it has been gathered in `self.results`